fml/backtrace.cc

```diff
@@ -55,10 +61,18 @@ static std::string GetSymbolName(void* symbol) {
   return DemangleSymbolName({name});
 }
 
+static int Backtrace(void** symbols, int size) {
+#if FML_OS_WIN
+  return CaptureStackBackTrace(0, size, symbols, NULL);
+#else
+  return ::backtrace(symbols, size);
+#endif  // FML_OS_WIN
+}
+
 std::string BacktraceHere(size_t offset) {
   constexpr size_t kMaxFrames = 256;
   void* symbols[kMaxFrames];
-  const auto available_frames = ::backtrace(symbols, kMaxFrames);
+  const auto available_frames = Backtrace(symbols, kMaxFrames);
```

Comment:
Should/could we use [abseil's `GetStackTrace`](https://github.com/abseil/abseil-cpp/blob/a05366d851c5cb88065272f951e03955197e7c11/absl/debugging/stacktrace.h#L133) here?

Reply:
It looks nice! I'm replacing with this. I'll check output on linux...

Reply:
I also find `absl::InstallFailureSignalHandler`. The engine doesn't use `BacktraceHere` out of signal handler, so we can completely replace backtrace.cc with `absl::InstallFailureSignalHandler`.

cc: @chinmaygarde @cbracken

Reply:
Source code (prototype): https://github.com/moko256/flutter_engine/commit/3ca3f107b19bfd21256657fbab9ab08954c96a20
Windows:
```
*** SIGABRT received at time=1663690051 ***
    @   000000014073933C  (unknown)  absl::AbslFailureSignalHandler
    @   0000000142957225  (unknown)  raise
    @   00000001428D5DE9  (unknown)  abort
    @   00000001400E74A9  (unknown)  fml::testing::BacktracePrinter
    @   00000001400E748E  (unknown)  fml::testing::BacktraceTest_BacktraceHere_Test::TestBody
    @   00000001427BA210  (unknown)  testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test,void>
    @   00000001427A4201  (unknown)  testing::internal::HandleExceptionsInMethodIfSupported<testing::Test,void>
    @   000000014278A59F  (unknown)  testing::Test::Run
    @   000000014278ADA8  (unknown)  testing::TestInfo::Run
    @   000000014278B41C  (unknown)  testing::TestSuite::Run
    @   00000001427956E5  (unknown)  testing::internal::UnitTestImpl::RunAllTests
    @   00000001427BE6F0  (unknown)  testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl,bool>
    @   00000001427A6E91  (unknown)  testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl,bool>
    @   0000000142795274  (unknown)  testing::UnitTest::Run
    @   0000000140279BC1  (unknown)  RUN_ALL_TESTS
    @   0000000140279848  (unknown)  main
    @   00000001428636A9  (unknown)  invoke_main
    @   000000014286358E  (unknown)  __scrt_common_main_seh
    @   000000014286344E  (unknown)  __scrt_common_main
    @   000000014286373E  (unknown)  mainCRTStartup
    @   00007FFFC2AB7034  (unknown)  BaseThreadInitThunk
    @   00007FFFC3A426A1  (unknown)  RtlUserThreadStart
```

Linux:
```
*** SIGABRT received at time=1663690758 on cpu 0 ***
PC: @     0x7f11f6efb7bb  (unknown)  raise
    @     0x55b6704b8eec         64  (unknown)
    @     0x55b6704b8c02         64  (unknown)
    @     0x7f11f7219730  1921871696  (unknown)
    @     0x55b66ff31941         32  (unknown)
    @     0x55b67205407b         96  (unknown)
    @     0x55b672046c27        112  (unknown)
    @     0x55b672036073         96  testing::Test::Run()
    @     0x55b672036734        112  testing::TestInfo::Run()
    @     0x55b672036d3d        112  testing::TestSuite::Run()
    @     0x55b6720404e1        208  testing::internal::UnitTestImpl::RunAllTests()
    @     0x55b6720571fb         96  (unknown)
    @     0x55b672048617        112  (unknown)
    @     0x55b6720400df         96  testing::UnitTest::Run()
    @     0x55b6700b8961         16  (unknown)
    @     0x55b6700b8832        720  (unknown)
    @     0x7f11f6ee809b  (unknown)  __libc_start_main
    @ 0x5541d68949564100  (unknown)  (unknown)
Aborted
```